azurerm/resource_arm_autoscale_setting.go

```diff
@@ -0,0 +1,954 @@
+}
+
+func flattenAzureRmAutoscaleSchedule(recurrence insights.Recurrence) *schema.Set {
+	schedule := make(map[string]interface{})
+	schedule["time_zone"] = *recurrence.Schedule.TimeZone
+
+	days := make([]interface{}, len(*recurrence.Schedule.Days))
+	for k, v := range *recurrence.Schedule.Days {
+		days[k] = v
+	}
+	schedule["days"] = days
+
+	hours := make([]interface{}, len(*recurrence.Schedule.Hours))
+	for k, v := range *recurrence.Schedule.Hours {
+		hours[k] = v
+	}
+	schedule["hours"] = hours
+
+	minutes := make([]interface{}, len(*recurrence.Schedule.Minutes))
+	for k, v := range *recurrence.Schedule.Minutes {
+		minutes[k] = v
+	}
+	schedule["minutes"] = minutes
+
+	return schema.NewSet(resourceAzureRmAutoscaleDefaultHash, []interface{}{schedule})
+}
+
+func flattenAzureRmAutoscaleNotification(notifications *[]insights.AutoscaleNotification) []interface{} {
+	results := make([]interface{}, 0, len(*notifications))
+
+	for _, notification := range *notifications {
+		notificationConfig := make(map[string]interface{})
+		notificationConfig["operation"] = *notification.Operation
+		notificationConfig["email"] = flattenAzureRmAutoscaleEmailNotification(notification)
+		if *notification.Webhooks != nil {
+			notificationConfig["webhook"] = flattenAzureRmAutoscaleWebhook(notification)
+		}
+		results = append(results, notificationConfig)
+	}
+
+	return results
+}
+
+func flattenAzureRmAutoscaleEmailNotification(notification insights.AutoscaleNotification) *schema.Set {
+	email := make(map[string]interface{})
+	email["send_to_subscription_administrator"] = *notification.Email.SendToSubscriptionAdministrator
+	email["send_to_subscription_co_administrator"] = *notification.Email.SendToSubscriptionCoAdministrators
+
+	if *notification.Email.CustomEmails != nil {
+		email["custom_emails"] = *notification.Email.CustomEmails
+	}
+
+	return schema.NewSet(resourceAzureRmAutoscaleDefaultHash, []interface{}{email})
+}
+
+func flattenAzureRmAutoscaleWebhook(notification insights.AutoscaleNotification) *schema.Set {
+	set := &schema.Set{
+		F: resourceAzureRmAutoscaleDefaultHash,
+	}
+
+	for _, v := range *notification.Webhooks {
+		webhook := map[string]interface{}{}
+		webhook["service_uri"] = *v.ServiceURI
+
+		if *v.Properties != nil {
+			properties := map[string]interface{}{}
+			for key, value := range *v.Properties {
+				properties[key] = *value
+			}
+			webhook["properties"] = properties
+		}
+		set.Add(webhook)
+	}
+
+	return set
+}
+
+func resourceAzureRmAutoscaleDefaultHash(v interface{}) int {
+	var buf bytes.Buffer
+	m := v.(map[string]interface{})
+	var keys []string
+
+	for k := range m {
+		keys = append(keys, k)
+	}
+	sort.Strings(keys)
+
+	for _, k := range keys {
+		buf.WriteString(fmt.Sprintf("%s:%v;", k, m[k]))
+	}
+
+	return hashcode.String(buf.String())
+}
+
+func listTimeZoneNames() []string {
+	return []string{
+		"Dateline Standard Time", "UTC-11", "Hawaiian Standard Time", "Alaskan Standard Time",
```

Comment:
is it possible to use a regex for ex. `UTC+11` here instead, rather than listing each of these out?